napari/layers/utils/text.py

```diff
@@ -0,0 +1,328 @@
+from typing import Tuple, Union
+
+import numpy as np
+
+from ._text_constants import TextMode, Anchor
+from ._text_utils import (
+    format_text_properties,
+    get_text_anchors,
+)
+from ...utils.colormaps.standardize_color import transform_color
+from ...utils.event import EmitterGroup, Event
+
+
+class TextManager:
+    """ Manages properties related to text displayed in conjunction with the layer
+
+    Parameters
+    ----------
+    text : array or str
+        the strings to be displayed
+    rotation : float
+        Angle of the text elements around the anchor point. Default value is 0.
+    anchor : str
+        The location of the text origin relative to the bounding box.
+        Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'
+    translation : np.ndarray
+        Offset from the anchor point.
+    color : array or str
+        Font color for the text
+    size : float
+        Font size of the text. Default value is 12.
+    visible : bool
+        Set to true of the text should be displayed.
+
+    Attributes
+    ----------
+    text : array
+        the strings to be displayed
+    rotation : float
+        Angle of the text elements around the anchor point. Default value is 0.
+    anchor : str
+        The location of the text origin relative to the bounding box.
+        Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'.
+    translation : np.ndarray
+        Offset from the anchor point.
+    color : array
+        Font color for the text
+    size : float
+        Font size of the text. Default value is 12.
+    visible : bool
+        Set to true of the text should be displayed.
+    """
+
+    def __init__(
+        self,
+        text,
+        n_text,
+        properties={},
+        rotation=0,
+        translation=0,
+        anchor='center',
+        color='black',
```

Comment:
This will be controversial, but I'm thinking that maybe we should consider using cyan as the default text colour. I'm finding that cyan on white looks reasonable, but black on black (most fluorescent images) is unusable.

Reply:
I agree that black is definitely not the right choice. It's cyan now. We can revise it later if there is an uprising 😛

Reply:
> I agree that black is definitely not the right choice. It's cyan now. We can revise it later if there is an uprising 😛

I would consider our hightlight color - which I find a bit easier on the eyes then cyan. One downside though is it is the highlight color and so that might be confusing too .... cyan is fine for now